### ISSUE TITLE
MacOS multiple monitors fail

### ISSUE THREAD

**Issue by waynenilsen** [maps4print/azul#39]
## Description

Running calculator `cargo run --example calculator` from current master, I get a white screen when I move to my secondary monitor. 

## Version / OS

* azul version: 0.1.0

* Operating system: MacOS 10.13.6

* Windowing system (X11 or Wayland, Linux only): N/A

## Steps to Reproduce

* In MacOS 10.13.6 with rustc 1.30.0 (da5f414c2 2018-10-24) clone this repo at  3950674 .
* Run `cargo run --example calculator`
* Move window to secondary monitor. 

## Expected Behavior

Calculator is visible

## Actual Behavior

Blank screen is seen 

## Additional Information

I love this project, good luck!

**Comment by fschutt** [comment]
Yeah, also see #5 - macOS is currently very broken. I currently can't debug macOS related bugs since I don't have a Mac.

**Comment by waynenilsen** [comment]
That makes sense. I see that at one point you had [travis for OSX enabled](https://github.com/maps4print/azul/blob/master/.travis.yml#L9) that may be a way out of the issue of not having a mac to test on?

**Comment by fschutt** [comment]
No, that just runs cargo test, doesn't test anything graphically. I can't detect rendering issues automatically.

**Comment by neongreen** [comment]
> not having a Mac to test on

Will something like https://portal.macincloud.com/select/#/plans cover this need? They provide RDP access to a macOS box for $20/mo. I would be happy to sponsor that.

**Comment by fschutt** [comment]
Yeah, I've thought about this, I recently read a blog entry of people running things via RDP. I think it's a good solution, but besides not having a Mac, I sadly have the problem of having too little time right now. In the end, I only made azul to write a desktop application, and I need to get that finished, then I'll have more time.

**Comment by dataf3l** [comment]
@fschutt I have a macos, if you need me to do any testing, just say the word, I'd be happy to help in any way I can, with bug reports and stuff.

**Comment by dataf3l** [comment]
Also, mac is nice, maybe getting one?

**Comment by fschutt** [comment]
Yeah @dataf3l if you have a mac, read through #5 and try to debug it - it has something to do with webrender not re-rendering properly on resizing. I can't really help with debugging, though.

**Comment by schulzch** [comment]
I tried to reproduce this bug by running the calculator example today using my MBP and an external TV connected via HDMI. Running the calculator example works. Also, moving the window from the HiDPI display (MBP) to the LoDPI display (TV) seems to work.

I guess the initial scaling issues mentioned in [#50]( https://github.com/maps4print/azul/issues/50#issuecomment-462083920) that seem to go away after resizing are probably related to the framebuffer sizing issue mentioned here [#18](https://github.com/maps4print/azul/issues/18#issuecomment-462076263).

Thus, I suggest to close this one (blank screen issue is gone).

**Comment by fschutt** [comment]
I also think I know why it's happening: Azul currently takes the HiDPI factor from the first monitor it finds (which is usually the "primary" monitor), not the monitor that the actual window is going to appear on. So if your primary monitor happens to be a TV, but the "first" monitor is the MBP screen, then the DPI factor might get screwed. Need to investigate this (which monitor shows up as the "primary" and which one shows up as the "first").

**Comment by yay** [comment]
@fschutt I just tested this and can confirm your guess. If I make the external monitor my primary monitor, the scaling is correct. I do still get a black window on startup every once in a while, like in that gif I attached to another issue: ![out](https://user-images.githubusercontent.com/1799756/52526814-e26a9900-2ccf-11e9-8944-6aa4ae00e5c8.gif) Notice that when I run the app the 4th or 5th time, the window is black. This happens regardless of the display, but only about 20% of the time.
